qtranslate_core.php

```diff
@@ -447,6 +447,13 @@ function qtranxf_detect_language_front( &$url_info ) {
 function qtranxf_setcookie_language( $lang, $cookie_name, $cookie_path ) {
     global $q_config;
 
+    // fix "Headers already sent" error in PHP
+    // 'setcookie' is function modifying the HTTP header (need to send a 'Set-Cookie:')
+    // sometimes wp_cron.php SEEMS to be the source of headers being sent prematurely
+    // so we avoid setting the cookie during 'wp_doing_cron' if the headers have
+    // already been sent (it's painless for the plugin to skip this)
+    if(headers_sent() && wp_doing_cron()) return;
```

Comment:
It makes sense to check for `headers_sent()`.
But I'm wondering, do we even have to check for `wp_doing_cron()`?
If the headers are sent we're already in troubles in any case. We would see the "Headers already sent" as a debug message, but it's pretty bad anyway. This should be detected differently, perhaps with an assert or some other functionality that we don't need to solve now.

Reply:
But we might also be conservative and simply fix this case as it was already reported in #779.
Then we can consider a "headers already sent" warning as a debug message for the general case. It doesn't seem to happen anymore, at least from QTX. If other plugins echo something before it's not depending on us.

Reply:
Yes, I agree that `headers_sent()` is already enough.
But I inserted `wp_doing_cron()` for this reason:
I turned on debugging and controlled for a month. And all error messages were only when `wp_doing_cron()` was __true__.
If the headers have already been sent - it's an error. In another plugin or in the theme. And it means it should be seen and fixed. But `wp_cron.php` does not depend on us and won't be fixed. For me, it is logical to simply avoid this situation, especially since setting cookie during `wp_doing_cron()` there is no particular sense.

Reply:
OK, let's keep it as it is. Merged.